src/tests/profiler/unittest/releaseonshutdown.cs

```diff
@@ -0,0 +1,66 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace Profiler.Tests
+{
+    class ReleaseOnShutdown
+    {
+        static readonly Guid ReleaseOnShutdownGuid = new Guid("B8C47A29-9C1D-4EEA-ABA0-8E8B3E3B792E");
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void WasteTime()
+        {
+            // Give time for the profiler to detach
+            Console.WriteLine("Sleeping...");
+            for (int i = 0; i < 100; ++i)
+            {
+                Thread.Sleep(100);
```

Comment:
Timeouts are difficult in CI.  Is there a way to sync with the profiler to know when you can stop waiting, e.g., poll for a file the profiler creates on detach?  1 second might not be long enough in all CI scenarios, e.g., gcstress, jitstress, etc., and I'd rather not artificially increase the length of the test.

Reply:
I agree with John here - while adding the code to do this would be tedious, the time taken to diagnose a failure that happened because of such timeout is painful... And we learned this the hard way :-)